### PR TITLE
KIWI-1684 & KIWI-1695 refactor API tests

### PR DESCRIPTION
**Pull request by Caitlin-cooling**
### What changed

- Changed the way that we fetch the bodies of the TxMA events - do them all in one go and then validate each event against the list of all events
- Refactor the way we use ajv as the `require` was causing caching issues meaning that all events were being compared against one schema
- Removed retries on getting list of TxMA events because even one timeout was causing a timeout. With the validation changes this means that we just validate what we have and if an event is missing then that specific test will fail
- Pulled util functions out of ApiTestSteps and into an ApiUtil file. Also moved Api related files into the `/api` directory
- Moved the well known tests because they are not an e2e test

### Why did it change

To improve quality, make them easier to change in the future and prevent false positives

### Screenshots

If no events can be found for that session, tests fail
Screenshot 2024-03-01 at 11 20 37 am

If an event is expected but isn't found in the test harness, tests fail
Screenshot 2024-03-01 at 11 36 32 am

All tests pass as they have been written
image

### Issue tracking

- [KIWI-1684](https://govukverify.atlassian.net/browse/KIWI-1684)
- [KIWI-1695](https://govukverify.atlassian.net/browse/KIWI-1695)

### PII logging

- [x] Verified that no PII data is being logged

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

[KIWI-1684]: https://govukverify.atlassian.net/browse/KIWI-1684?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/tests/api/ApiTestSteps.ts [hunk 1]
```diff
@@ -181,7 +171,6 @@ export async function getSessionById(sessionId: string, tableName: string): Prom
 		session = Object.fromEntries(
 			Object.entries(originalSession).map(([key, value]) => [key, value.N ?? value.S]),
 		) as unknown as ISessionItem;
-		console.log("getSessionById Response", session);
```
Comment (ViveakV) [inline src/tests/api/ApiTestSteps.ts:184]: I know @Z-Mohamoud and @MohamedHassan101 leave logs in places to figure out why tests failing so just highlighting incase they want this kept

**Review by ViveakV — APPROVED** [review]

**Comment by ViveakV** [comment]
Nice one! 🥳

File: src/tests/api/ApiTestSteps.ts (hunk 1 above)
Comment (Caitlin-cooling) [reply src/tests/api/ApiTestSteps.ts:184]: Yeah happy to leave essential logs but they are pretty noisy so if there's anything we _can_ remove I'd like to do that. Let me know your thoughts Zac and Mohamed

Comment (Z-Mohamoud) [reply src/tests/api/ApiTestSteps.ts:184]: It's not that useful so happy for it to be removed

**Review by Z-Mohamoud — APPROVED** [review]

**Pull request merged**